argoverse/utils/make_att_files.py

```diff
@@ -0,0 +1,380 @@
+            if len(ind_close) > 0:
+                ind_close_max = ind_close.max() + 1
+                ind_close_min = ind_close.min()
+
+            if dict_tracks[id_track]["length_track"] == 0:
+                # this shouldn't happen
+                dict_tracks[id_track]["difficult_att"].append("zero_length")
+
+            else:
+                if dict_tracks[id_track]["list_dist"][ind_valid].min(
+                ) > dist_close:
+                    dict_tracks[id_track]["difficult_att"].append("far")
+                else:
+                    if dict_tracks[id_track][
+                            "length_track"] < 10 or dict_tracks[id_track][
+                                "exists"].sum() < 10:
+                        dict_tracks[id_track]["difficult_att"].append("short")
+                    else:
+                        if (ind_close_max - ind_close_min
+                            ) - dict_tracks[id_track]["exists"][
+                                ind_close_min:ind_close_max].sum() > 20:
+                            dict_tracks[id_track]["difficult_att"].append(
+                                "occ")
+
+                        if np.quantile(vel_abs[ind_valid][ind_close], 0.9) > 1:
+                            dict_tracks[id_track]["difficult_att"].append(
+                                "fast")
+
+                if len(dict_tracks[id_track]["difficult_att"]) == 0:
+                    dict_tracks[id_track]["difficult_att"].append("easy")
+
+        if visualize:
+            for ind_lidar, timestamp_lidar in enumerate(list_lidar_timestamp):
+
+                list_bboxes = []
+                list_difficulty_att = []
+
+                for id_track in dict_tracks.keys():
+                    if dict_tracks[id_track]["exists"][ind_lidar]:
+                        list_bboxes.append(
+                            dict_tracks[id_track]["list_bbox"][ind_lidar])
+                        list_difficulty_att.append(
+                            dict_tracks[id_track]["difficult_att"])
+
+                path_lidar = os.path.join(path_log, "lidar",
+                                          "PC_%s.ply" % timestamp_lidar)
+                egovehicle_to_city_se3 = argoverse_loader.get_pose(
+                    ind_lidar, id_log)
+                pc = np.asarray(o3d.io.read_point_cloud(path_lidar).points)
+                list_lidar_timestamp = data_log.lidar_timestamp_list
+                save_bev_img(path_output_vis, list_bboxes, list_difficulty_att,
+                             "argoverse_%s" % name_folder, id_log,
+                             timestamp_lidar, pc, egovehicle_to_city_se3)
+
+        for id_track in dict_tracks.keys():
+            list_key = list(dict_tracks[id_track].keys()).copy()
+            for key in list_key:
+                if key != "difficult_att":
+                    del dict_tracks[id_track][key]
+
+        dict_att_all[name_folder][id_log] = dict_tracks
+
+pickle_out = open(filename_output, "wb")
```

Comment:
we have this function in the api too, can we call it as a library function?

`from argoverse.utils.pkl_utils import save_pkl_dictionary`